src/main/java/org/opentripplanner/transit/service/TransitService.java

```diff
@@ -215,4 +215,16 @@ List<TripTimeOnDate> stopTimesForPatternAtStop(
    * So, if more patterns of mode BUS than RAIL visit the stop, the result will be [BUS,RAIL].
    */
   List<TransitMode> getModesOfStopLocation(StopLocation stop);
+
+  /**
+   * Iterates over all child stops, the routes that visit this stop and return a de-duplicated list
+   * of their agencies.
+   */
+  List<Agency> getAgenciesForStopLocationsGroup(StopLocationsGroup group);
+
+  /**
+   * Iterates over all routes that visit this stop location and return a de-duplicated list
+   * of their agencies.
+   */
+  List<Agency> getAgenciesForStopLocation(StopLocation stop);
 }
```

Comment:
Are these methods commonly used or use-case specific? The name do not align with the JavaDoc - in the JavaDoc you talk about Rotes, but that is not part of the method name. A well designed domain service expose generic logic which can be reused. The generic part of this is the relationship between group -> stop -> route, the specific is the filtering of route agency to get get a distinct list of agency-ids.

Reply:
At the moment they are only used in one place but I would like to add this kind of shortcut to the API as well.

It's answering the common question of "what agencies use a stop?". Since there isn't a direct relationship between a stop and an agency, you have to go the detour via the routes. But maybe that is too implementation specific and should not be explicitly mentioned in the Javadoc.

Reply:
The `Route` is not an implementation detail, there are other relationships which link a stop-location and an agency. For example the agency responsible for maintaining its physical infrastructure. So, you should include the fact that this is the agencies using the stop for transit. 

When shortcutting relationship in a model it is important that there is not ambiguity. If there is a natural relationship it is normally safe to do so, if not it should be avoided because it leads to errors. For example in this case the set of agencies you get iterating over the trips visiting the stop and the routes are not the same. In some cases you want the trips and in other the routes would be ok.

Reply:
Can a trip have several routes and therefore several agencies?

What is your request? Should I move this into my use-case specific code?

Reply:
It can only have one Route, but it may have a different Operator than the route - leading to different agency. This is common in the real world, but one of the cases GTFS does not cover.

Reply:
Ok, would you like me to just update the Javadoc or move this method to a usecase specific class?

Reply:
They do not belong in a domain service - there is no general business rule, so they should be inlined/moved to the place where they are used (use-case specific code).

Reply:
I've done that in https://github.com/opentripplanner/OpenTripPlanner/pull/5634/commits/9f66f5db54ed12a6b426d50b2f7f5acb1c7db1d2